src/verify.js

```diff
@@ -0,0 +1,68 @@
+const utils = require('./utils');
+
+const verifyViews = async (dbName, numDocs) => {
+  if (!numDocs) {
+    return true;
+  }
+
+  const url = utils.getUrl(`${dbName}/_design_docs`, false, 'include_docs=true');
+  const response = await utils.request({ url });
+
+  let viewsIndexed = false;
+  let viewsChecked = false;
```

Comment:
If I'm reading this correctly, the only time this will stay false is if, there are no ddocs or no views, right? It would be more readable to short circuit this special case, eg:

```suggestion
  const ddocsWithViews = response.rows.filter(row => row.ddoc && row.ddoc.views && row.ddoc.views.length !== 0);
  if (!ddocsWithViews.length) {
    return true; // nothing to verify
  }
  for (const { doc: ddoc } of ddocsWithViews) {
```

Obviously I've just hacked this together - ignore typos and obvious bugs!

If you do this and the other short circuit mentioned below you can drop both variables and just `return false` at the end, which makes the code easier to follow because you don't need to track all the ways in which the variable can change.